pkg/vpn_server/values.go

```diff
@@ -48,9 +48,28 @@ func BuildValues(cfg config.VPNServer) (openvpn.SeedServerValues, error) {
 		v.OpenVPNNetwork = cfg.VPNNetwork
 
 		v.SeedPodNetwork = cfg.SeedPodNetwork
-		v.ShootNetworks = append(v.ShootNetworks, network.ParseIPNet(constants.ShootNodeNetworkMapped))
-		v.ShootNetworks = append(v.ShootNetworks, network.ParseIPNet(constants.ShootSvcNetworkMapped))
-		v.ShootNetworks = append(v.ShootNetworks, network.ParseIPNet(constants.ShootPodNetworkMapped))
+		// v4 networks are mapped to 240/4, v6 networks are kept as is
+		for _, serviceNetwork := range cfg.ServiceNetworks {
+			if serviceNetwork.IP.To4() != nil {
+				v.ShootNetworks = append(v.ShootNetworks, network.ParseIPNet(constants.ShootSvcNetworkMapped))
+			} else {
+				v.ShootNetworks = append(v.ShootNetworks, serviceNetwork)
+			}
+		}
+		for _, podNetwork := range cfg.PodNetworks {
+			if podNetwork.IP.To4() != nil {
+				v.ShootNetworks = append(v.ShootNetworks, network.ParseIPNet(constants.ShootPodNetworkMapped))
+			} else {
+				v.ShootNetworks = append(v.ShootNetworks, podNetwork)
+			}
+		}
+		for _, nodeNetwork := range cfg.NodeNetworks {
+			if nodeNetwork.IP.To4() != nil {
+				v.ShootNetworks = append(v.ShootNetworks, network.ParseIPNet(constants.ShootNodeNetworkMapped))
+			} else {
+				v.ShootNetworks = append(v.ShootNetworks, nodeNetwork)
+			}
+		}
```

Comment:
We probably do not want to add the mapped ranges more than once. Should we ensure this and handle the error case properly?

Reply:
I've added a commit that removes duplicates in the shoot networks as well as a test for it.